src/python/tests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,234 @@
+            tmp_dir: Unit test temp directory (fixture).
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            orthology_benchmark.dump_genomes(species_list, species_set_name, self.host, self.port,
+                                             self.username, tmp_dir)
+
+            out_files = tmp_dir / species_set_name
+            # pylint: disable-next=no-member
+            exp_out = pytest.files_dir / "orth_benchmark"  # type: ignore[attr-defined]
+            for db_name, unittest_db in self.core_dbs.items():
+                assert file_cmp(out_files / f"{unittest_db.dbc.db_name}.fasta", exp_out / f"{db_name}.fasta")
+
+    def test_dump_genomes_fake_connection(self, tmp_dir: Path) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` with fake server details.
+
+        Args:
+            tmp_dir: Unit test temp directory (fixture).
+
+        """
+        with raises(sqlalchemy.exc.OperationalError):
+            orthology_benchmark.dump_genomes(["mus_musculus", "naja_naja"], "default",
+                                             "fake-host", 65536, "compara", tmp_dir)
+
+
+@pytest.mark.parametrize(
+    "core_names, exp_output, expectation",
+    [
+        (["mus_musculus_core_105_1", "mus_musculus_core_52_105_3", "mus_musculus_core_104_4"],
+         "mus_musculus_core_52_105_3", does_not_raise()),
+        ([], None, raises(ValueError,
+                          match=r"Empty list of core databases. Cannot determine the latest one."))
+    ]
+)
+def test_find_latest_core(core_names: List[str], exp_output: str, expectation: ContextManager) -> None:
+    """Tests :func:`orthology_benchmark.find_latest_core()` function.
+
+    Args:
+        core_names: A list of core database names.
+        exp_output: Expected return value of the function.
+        expectation: Context manager for the expected exception, i.e. the test will only pass if that
+            exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+    """
+    with expectation:
+        assert orthology_benchmark.find_latest_core(core_names) == exp_output
+
+
+@pytest.mark.parametrize(
+    "multi_dbs",
+    [
+        [{'src': 'core/danio_rerio_core_105_11', 'name': 'danio_rerio_core_105_11'},
+         {'src': 'core/mus_musculus_cbaj_core_107_1', 'name': 'mus_musculus_cbaj_core_107_1'},
+         {'src': 'core/mus_musculus_core_106_39', 'name': 'mus_musculus_core_106_39'}]
+    ],
+    indirect=True
+)
+class TestGetCoreNames:
+    """Tests :func:`orthology_benchmark.get_core_names()` function.
+
+    Attributes:
+        core_dbs: A set of test core databases.
+
+    """
+
+    core_dbs = {}  # type: Dict
+    host = None  # type: str
+    port = None  # type: int
+    username = None  # type: str
```

Comment:
Like before, this suggestion is entirely optional:
```suggestion
    host = None  # type: Optional[str]
    port = None  # type: Optional[int]
    username = None  # type: Optional[str]
```